### DISCUSSION TITLE
How to implement multiselect?

### DISCUSSION THREAD

**Discussion by shayani** [josefarias/hotwire_combobox#212] [Q&A]
I found the PR for multiselect, and also say that in the roadmap for version 1.0.0, the documentation for multiselect is on the list. But, I wasn't able to figure out how to implement multiselect.

I want to multiselect from a static list that doesn't have a direct relationship (`has_many`) to a model.

Thanks!

**Comment by ghost** [comment]
I tried to render a multiselect by looking at the tests. The only requirement I found is setting the attribute `multiselect_chip_src`. Yet it does not show up as a multiselect for me! 

These are the files I looked at: 
- [test/dummy/app/views/comboboxes/multiselect.html.erb](https://github.com/josefarias/hotwire_combobox/blob/c280da0f4b61386729959a28f32fbed728a0c773/test/dummy/app/views/comboboxes/multiselect.html.erb#L3)
- [test/dummy/app/views/state_chips/create_html.turbo_stream.erb](https://github.com/josefarias/hotwire_combobox/blob/c280da0f4b61386729959a28f32fbed728a0c773/test/dummy/app/views/state_chips/create_html.turbo_stream.erb)

What am I missing? Is `multiselect_chip_src` the only option required to use a multiselect?

In my case I have the `has_many` in place.

**Reply by josefarias** [reply]
Hey @Aiderlei. Possibly a silly question, but you mention `multiselect_chip_scr`, is it possible you misspelled `multiselect_chip_src` in your code?

Be aware the multiselect API is in its early stages so it's subject to change, which is why it's not documented yet.

I'm hesitant to provide definitive instructions here for the same reason (cc @shayani).

**Reply by ghost** [reply]
Hi @josefarias, thanks for the quick response!

The misspelling was only in my question, not in my code. However, I’m starting to think the returned value for multiselect_chip_src might not meet the requirements. That said, I’m still struggling to understand how just setting multiselect_chip_src would be sufficient to render a multiselect.

Thank you for pointing that out!

I understand that the multiselect API is still in its early stages and subject to change. I’ll make sure to keep an eye on future updates and documentation.

Appreciate the help! 😊
